Sources/SwiftSyntaxBuilderGeneration/Templates/BuildableCollectionNodesFile.swift

```diff
@@ -0,0 +1,257 @@
+        ].map { .docLineComment($0) + .newline }.reduce([], +),
+        modifiers: [TokenSyntax.public],
+        parameters: ParameterClause(
+          parameterList: [
+            FunctionParameter(
+              firstName: .wildcard,
+              secondName: .identifier("elements"),
+              colon: .colon,
+              type: ArrayType(elementType: elementType.expressibleAs)
+            )
+          ]
+        )
+      ) {
+        SequenceExpr(elements: ExprList([
+          MemberAccessExpr(base: "self", name: "elements"),
+          AssignmentExpr(),
+          elementType.isToken
+            ? "elements"
+            : FunctionCallExpr(MemberAccessExpr(base: "elements", name: "map"), trailingClosure: ClosureExpr {
+                FunctionCallExpr(MemberAccessExpr(base: "$0", name: "create\(elementType.buildableBaseName)"))
+              }),
+        ]))
+      }
+
+      InitializerDecl(
+        leadingTrivia: .docLineComment("/// Creates a new `\(type.buildable)` by flattening the elements in `lists`") + .newline,
+        modifiers: [TokenSyntax.public],
+        parameters: ParameterClause(
+          parameterList: [
+            FunctionParameter(
+              firstName: .identifier("combining").withTrailingTrivia(.space),
+              secondName: .identifier("lists"),
+              colon: .colon,
+              type: ArrayType(elementType: type.expressibleAs)
+            )
+          ]
+        )
+      ) {
+        SequenceExpr(elements: ExprList([
+          "elements",
+          AssignmentExpr(),
+          FunctionCallExpr(MemberAccessExpr(base: "lists", name: "flatMap"), trailingClosure: ClosureExpr {
+            CodeBlockItem(item: MemberAccessExpr(
+              base: FunctionCallExpr(MemberAccessExpr(base: "$0", name: "create\(type.buildable)")),
+              name: "elements"
+            ))
+          }),
+        ]))
+      }
+
+      InitializerDecl(
+        modifiers: [TokenSyntax.public],
+        parameters: ParameterClause(
+          parameterList: [
+            FunctionParameter(
+              firstName: .identifier("arrayLiteral").withTrailingTrivia(.space),
+              secondName: .identifier("elements"),
+              colon: .colon,
+              type: elementType.expressibleAs,
+              ellipsis: .ellipsis
+            )
+          ]
+        )
+      ) {
+        FunctionCallExpr(MemberAccessExpr(base: "self", name: "init")) {
+          TupleExprElement(expression: "elements")
+        }
+      }
+
+      FunctionDecl(
+        modifiers: [TokenSyntax.public],
+        identifier: .identifier("build\(type.baseName)"),
+        signature: FunctionSignature(
+          input: createFormatLeadingTriviaParameters(withDefaultTrivia: true),
+          output: type.syntax
+        )
+      ) {
+        VariableDecl(letOrVarKeyword: .let) {
+          PatternBinding(
+            pattern: "result",
+            initializer: InitializerClause(
```

Comment:
Perhaps we could add an optional parameter or a new convenience initializer in `VariableDeclConvenienceInitializers` for variable declarations binding an initializer clause? This seems like a common use case and might be worth sugaring.

Reply:
-> #523